lib/SPIRV/OCLUtil.cpp

```diff
@@ -143,10 +143,21 @@ unsigned getOCLVersion(Module *M) {
   NamedMDNode *NamedMD = M->getNamedMetadata(kSPIR2MD::OCLVer);
   if (!NamedMD)
     return 0;
-  assert (NamedMD->getNumOperands() == 1 && "Invalid SPIR");
-  MDNode *MD = NamedMD->getOperand(0);
-  unsigned Major = getMDOperandAsInt(MD, 0);
-  unsigned Minor = getMDOperandAsInt(MD, 1);
+  assert (NamedMD->getNumOperands() > 0 && "Invalid SPIR");
```

Comment:
It is not so useful to fix the metadata since the fixed metadata won't be used by any passes. Instead new SPIR-V metadata with different format are generated and used by later passes. In the case of OpenCL version, the original function provides the required information. The modification is to check integrity in case multiple operands exist but the semantic of the original function does not change.

Thanks for the suggested code. I will fix that.